backend/src/crd/controller/scheduledworkflow/util/cron_schedule.go

```diff
@@ -42,41 +43,41 @@ func NewCronSchedule(cronSchedule *swfapi.CronSchedule) *CronSchedule {
 
 // GetNextScheduledEpoch returns the next epoch at which a workflow must be
 // scheduled.
-func (s *CronSchedule) GetNextScheduledEpoch(lastJobEpoch *int64,
-	defaultStartEpoch int64) int64 {
-	effectiveLastJobEpoch := s.getEffectiveLastJobEpoch(lastJobEpoch, defaultStartEpoch)
-	return s.getNextScheduledEpoch(effectiveLastJobEpoch)
+func (s *CronSchedule) GetNextScheduledEpoch(lastJobTime *v1.Time,
+	defaultStartTime time.Time, location *time.Location) int64 {
+	effectiveLastJobTime := s.getEffectiveLastJobEpoch(lastJobTime, defaultStartTime, location)
+	return s.getNextScheduledEpoch(effectiveLastJobTime, location)
 }
 
-func (s *CronSchedule) GetNextScheduledEpochNoCatchup(lastJobEpoch *int64,
-	defaultStartEpoch int64, nowEpoch int64) int64 {
+func (s *CronSchedule) GetNextScheduledEpochNoCatchup(lastJobTime *v1.Time,
+	defaultStartTime time.Time, nowTime time.Time, location *time.Location) int64 {
 
-	effectiveLastJobEpoch := s.getEffectiveLastJobEpoch(lastJobEpoch, defaultStartEpoch)
-	return s.getNextScheduledEpochImp(effectiveLastJobEpoch, false, nowEpoch)
+	effectiveLastJobTime := s.getEffectiveLastJobEpoch(lastJobTime, defaultStartTime, location)
+	return s.getNextScheduledEpochImp(effectiveLastJobTime, false, nowTime, location)
 }
 
-func (s *CronSchedule) getEffectiveLastJobEpoch(lastJobEpoch *int64,
-	defaultStartEpoch int64) int64 {
+func (s *CronSchedule) getEffectiveLastJobEpoch(lastJobTime *v1.Time,
+	defaultStartTime time.Time, location *time.Location) time.Time {
 
 	// Fallback to default start epoch, which will be passed the Job creation
 	// time.
-	effectiveLastJobEpoch := defaultStartEpoch
-	if lastJobEpoch != nil {
+	effectiveLastJobTime := defaultStartTime
+	if lastJobTime != nil {
 		// Last job epoch takes first precedence.
-		effectiveLastJobEpoch = *lastJobEpoch
+		effectiveLastJobTime = lastJobTime.Time
 	} else if s.StartTime != nil {
 		// Start time takes second precedence.
-		effectiveLastJobEpoch = s.StartTime.Unix()
+		effectiveLastJobTime = s.StartTime.In(location)
```

Comment:
Can you help me understand why we need a `.In(location)` here?
A time is a time in a timezone, no matter which timezone it sits in, adding the location info here doesn't affect the time instant, only the location representation.

Reply:
yes you are correct, it is not needed.